webapp/player-project/src/ws-proxy.ts

```diff
@@ -38,6 +38,15 @@ const WebSocketProxy = new Proxy(window.WebSocket, {
         }
         return Reflect.set(target, prop, value);
       },
+      get(target, prop, receiver) {
+        const value = Reflect.get(target, prop, receiver);
+        // Bind native methods to the original WebSocket instance
+        // Websocket send and other method need the `this` to points to the WebSocket instance
```

Comment:
```suggestion
        // Because these methods are part of the native WebSocket prototype,       
        // they must be called with the original WebSocket as `this`.
        // If they're called with the Proxy as `this`, it results in an "illegal invocation".
        // Binding them to the underlying `target` (the real WebSocket) avoids this issue.
```